### ISSUE TITLE
7.2.0 Release Checklist

### ISSUE THREAD

**Issue by aaronayres35** [enthought/pyface#823]
Target date:

Release Candidate by YYYY-MM-DD

Release by YYYY-MM-DD

Release candidate
-------------------
- [x] Verify that all relevant PRs have been merged to master.
- [x] Create a PR against master to bump version number, merge that PR (#824)
- [x] From the commit just before bumping the version, create a new branch `maint/<release version number>`
- [x] Update changelog and open PR targeting a new `maint/<release version number>` branch (#825)
- [x] Update `ci-src-requirements.txt` if needed
- [x] Check MANIFEST and requirements are still up to date.
- [x] Update version in setup.py for the prerelease, open 2 PRs against `maint/<release version number>`
  - [x] Create a new branch from `maint/<release version number>`: `git checkout maint/<release version number>`; `git pull`; `git checkout -b call-it-anything-you-like`
  - [x] Set `PRERELEASE` to "rc1" and `IS_RELEASED` to true, commit, e.g. `git commit -m "Set IS_RELEASED to true for prerelease <release version number>rc1"`
  - [x] Open and merge a PR against `maint/<release version number>` (#826)
  - [x] Create a new branch from `maint/<release version number>`, Flip `IS_RELEASED` back to false, commit.
  - [x] Open and merge another PR against `maint/<release version number>` (#827)
- [x] Tag (annotated!) the release candidate on the commit where IS_RELEASED is set to true, e.g. `git tag -a -m "Release candidate <release version number>rc1" <release version number>rc1 <commit-hash>`
- [x] Push the tag to GitHub
- [x] Upload to PyPI
       - `git checkout <tag>`, `git clean -ffxd`, `python setup.py sdist`, `twine check dist/<...>.tar.gz`, `twine upload dist/<...>.tar.gz`
- [x] Announcement for the release candidate

Release blockers
----------------
- [x] (#830)

Pre-release
---
- [x] Backport PRs that have been merged to master to the maintenance branch. Use the "need backport ..." tag if there is one (but don't rely 100% on it) (#838)
- [x] Verify that no other open issue needs to be addressed before the release.
- [x] Test against other ETS packages and other ETS-using projects (done: traitsui, canopy data)
- [x] Check MANIFEST, requirements, changelog are still up to date.
- [x] Test building the documentation

Release
-------
- [x] Create branch `release/<release version number>` from `maint/<release version number>` branch.
  - [x] Set release to `<release version number>`, and set `IS_RELEASED` is true; commit
  - [x] Install from source distribution and run tests again
  - [x] Open a PR against `maint/<release version number>` with this being the last commit so that CI is built on the release commit (#839)
  - [x] Once CI is done building merge PR
  - [x] Bump the micro version number i.e. `<release version number + 0.0.1>` and set `IS_RELEASED` to false; commit.
  - [x] Open and then merge a separate PR against `maint/<release version number>` (#840)
- [x] From the commit at which `IS_RELEASED` is true and version is `<release version number>`, tag (annotated!) `git tag -a -m "Release <release version number>" <release version number>`
- [x] Push the tag `git push origin <release version number>`
- [x] Make PR targeting `gh-pages` branch: Generate documentation and copy the content to the branch. Verify that the resulting index.html looks good. (#841)
- [x] Upload to PyPI
- [x] Test the PyPI package

Post-release
-------------
- [x] Package update for `enthought/free` repository (for EDM)
- [x] Backport release note and change log to master, and possibly `maint/<release version number>` branch. (#848)
- [x] Update GitHub Release pages https://github.com/enthought/pyface/releases
- [x] Announcement (e.g. Google Group)

**Comment by aaronayres35** [comment]
When testing building the documentation I saw the following warnings:

```
/Users/aayres/Desktop/pyface/pyface/i_application_window.py:docstring of pyface.i_application_window.IApplicationWindow:7: WARNING: Unexpected section title.

Usage
-----
WARNING: autodoc: failed to import module 'api' from module 'pyface.preference'; the following exception was raised:
cannot import name 'PreferenceDialog'
/Users/aayres/Desktop/pyface/pyface/tasks/tasks_application.py:docstring of pyface.tasks.tasks_application.TaskFactory.factory:3: WARNING: Inline strong start-string without end-string.
/Users/aayres/Desktop/pyface/pyface/timer/do_later.py:docstring of pyface.timer.do_later.do_later:5: WARNING: Inline strong start-string without end-string.
/Users/aayres/Desktop/pyface/pyface/timer/do_later.py:docstring of pyface.timer.do_later.do_later:6: WARNING: Inline strong start-string without end-string.
/Users/aayres/Desktop/pyface/pyface/timer/do_later.py:docstring of pyface.timer.do_later.do_after:7: WARNING: Inline strong start-string without end-string.
/Users/aayres/Desktop/pyface/pyface/timer/do_later.py:docstring of pyface.timer.do_later.do_after:8: WARNING: Inline strong start-string without end-string.
/Users/aayres/Desktop/pyface/docs/source/toolkits.rst:65: WARNING: Unknown interpreted text role "py:exception".
looking for now-outdated files... none found
pickling environment... done
checking consistency... /Users/aayres/Desktop/pyface/docs/source/api/modules.rst: WARNING: document isn't included in any toctree
```

**Comment by aaronayres35** [comment]
> When testing building the documentation I saw the following warnings:

Note these warnings were present in 7.1.0 as well (they have not recently been introduced)

**Comment by aaronayres35** [comment]
closing as I have just made the announcement
